docs/architecture/blazor-for-web-forms-developers/introduction.md

```diff
@@ -8,6 +8,50 @@ ms.date: 09/11/2019
+
+Does this mean ASP.NET Web Forms is *dead* and should no longer be used? Of course not! As long as the .NET Framework continues to ship as part of Windows, it will be a supported framework. If you don't have a requirement for cross-platform support or open source or any of the other new features in .NET Core or .NET 5, then sticking with ASP.NET Web Forms on Windows is just fine. ASP.NET Web Forms development will be alive and well for many years to come.
+
+But there's another trend worth considering, and that's the shift to the client.
+
+## Client-side web development
+
+All of the various .NET based web frameworks, including ASP.NET Web Forms, have historically had one thing in common: they're *server-rendered*. In server-rendered web apps, the browser makes a request to the server, which then executes some code (.NET code in ASP.NET apps) to produce a response that is sent back to the browser to handle. In this model, the browser is used as a thin rendering engine. All of the hard work of producing the UI, running the business logic, and managing state happens on the server.
+
+However, browsers have become versatile platforms implementing an ever-increasing number of open web standards that grant access to the capabilities of the user's machine. Why not take advantage of the compute power, storage, memory, and other resources of the client device? UI interactions in particular can benefit from a richer and more interactive feel when handled at least partially or completely client-side. Logic and data that should be handled on the server can still be handled server-side using Web API calls or even over real-time protocols, like WebSockets. All of these benefits are available to web developers for free as long as they're willing to write some JavaScript. Client-side UI frameworks that simplify browser development, like Angular, React, and Vue, have grown in popularity. ASP.NET Web Forms developers can also benefit from leveraging the client, and even have some out of the box support with integrated JavaScript frameworks like ASP.NET AJAX.
+
+But having to bridge two different platforms and ecosystems (.NET and JavaScript) comes with a cost. You as the web developer have to become an expert in two parallel worlds with different languages, frameworks, and tools. Code and logic cannot be easily shared between client and server, resulting in duplication and engineering overhead. It can also be difficult to keep up with the JavaScript ecosystem, which has a history of evolving at breakneck speed, quickly changing preferences for different front-end frameworks (such as jQuery, Knockout, Angular, React, Vue, and so on) and build tools (such as grunt, gulp, webpack, and so on). But given JavaScript's browser monopoly there was little choice in the matter. That is until the web community got together and caused a *miracle* to happen!
```

Comment:
```suggestion
But having to bridge two different platforms and ecosystems (.NET and JavaScript) comes with a cost. You, as the web developer, have to become an expert in two parallel worlds with different languages, frameworks, and tools. Code and logic cannot be easily shared between client and server, resulting in duplication and engineering overhead. It can also be difficult to keep up with the JavaScript ecosystem, which has a history of evolving at breakneck speed, quickly changing preferences for different front-end frameworks (such as jQuery, Knockout, Angular, React, Vue, and so on) and build tools (such as grunt, gulp, webpack, and so on). But given JavaScript's browser monopoly, there was little choice in the matter. That is until the web community got together and caused a *miracle* to happen!
```